specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/examples/CreateHDInsightClusterWithAutoscaleConfig.json

```diff
@@ -2,7 +2,7 @@
   "parameters": {
     "clusterName": "cluster1",
     "resourceGroupName": "rg1",
-    "api-version": "2018-06-01-preview",
+    "api-version": "2015-03-01-preview",
```

Comment:
This change doesn't look right. 
And I think you should put the files for `2018-06-01-preview` under preview folder, not stable.

Reply:
> This change doesn't look right.
> And I think the files for `2018-06-01-preview` should be under preview folder, not stable.

api version is fixed.
And for why the file `2018-06-01-preview` under stable folder, this is a history problem.
This is intentional. It’s a little confusing, but per rule #5 here, whether the swagger spec is in the preview or stable folder “is not a direct analog for whether or not an API Version has the ‘-preview’ suffix or not. SDKs that are generated from 'preview' folder items should indicate to their customers in the most idiomatic way that breaking changes may still be coming.”

Basically the short version is that HDInsight is a GA service that still has an API with a -preview tag. Somebody made the decision several years ago to release stable versions of Hydra/Hyak SDKs for .NET, JavaScript, etc. that consumed this the -preview API. We are now releasing stable swagger-based SDKs so we can at least finally deprecate the Hydra/Hyak SDKs. As a result, we are making a commitment that we will not make breaking changes to this swagger spec version so we can GA some swagger-based SDKs.

Reply:
> This change doesn't look right.
> And I think the files for `2018-06-01-preview` should be under preview folder, not stable.

api version is fixed.
And for why the file `2018-06-01-preview` under stable folder, this is a history problem.